### ISSUE TITLE
+1 442 NPA-XXXX

### ISSUE THREAD

**Issue by kputland** [googlei18n/libphonenumber#1527]
Country/region affected (e.g., "US"): US
Example number(s) affected ("+1 555 555-1234"): +1 442 NXX-XXXX
The phone number range(s) to which the issue applies ("+1 555 555-XXXX"): +1 442 NXX-XXXX
The type of the number(s) ("fixed-line", "mobile", "short code", etc.): 
The cost, if applicable ("toll-free", "premium rate", "shared cost"):
Supporting evidence (for example, national numbering plan, announcement from mobile carrier, news article): http://www.allareacodes.com/442

**Comment by kkeshava** [comment]
Hi,
Thank you for taking the time to improve libphonenumber! We are already supporting the prefixes 283, 442, 458, 470, 475, 531, 534, 539, 681 and 762 in our library for the US. We will be updating only the geocoding details for them e.g 1283|Ohio. Please do let us know if you wanted us to fix anything else apart from geocoding as it is not very clear from your description. 
We've accepted this issue and will fix this in an upcoming release.

Please note that you have submitted this information with the knowledge that we may use, modify, reproduce, publicly display, and distribute the information in accordance with our open source policies. Please let us know if you did not intend to submit the information for such use.

**Comment by kputland** [comment]
Please help me find +1442 and the others...
I only submitted requests for NPA that I could not find in the geocoding data.

![image](https://cloud.githubusercontent.com/assets/1480163/22433062/4bec1064-e6d4-11e6-9ffd-5e730bff6cf9.png)

**Comment by kputland** [comment]
I would expect the following entries.

1442200|Calexico, CA
1442220|Nipton, CA
1442221|Benton, CA
1442222|Oceanside, CA
1442223|Newberry Springs, CA
1442224|Vista, CA
1442225|El Centro, CA
1442226|Brawley, CA
1442227|Palm Springs, CA
1442228|Bishop, CA
1442229|Victorville, CA
1442231|El Centro, CA
1442232|Carlsbad, CA
1442233|California
1442234|Palm Springs, CA
1442235|California
1442236|El Centro, CA
1442237|California
1442238|California
1442242|Victorville, CA
1442243|California
1442244|Carlsbad, CA
1442246|California
1442253|California
1442281|Escondido, CA
1442300|Indio, CA
1442321|California
1442333|Oceanside, CA
1442400|Indio, CA
1442444|Fallbrook, CA
1442500|Oceanside, CA
1442600|Hesperia, CA
1442666|Palm Springs, CA
1442777|Escondido, CA
1442800|Hesperia, CA
1442888|Oceanside, CA
1442900|El Centro, CA
1442999|Escondido, CA

**Comment by kputland** [comment]
I get non-specific "United States" from the geocoder
```
$ pip install -U phonenumbers
Collecting phonenumbers
  Downloading phonenumbers-8.1.0-py2.py3-none-any.whl (2.7MB)
    100% |████████████████████████████████| 2.7MB 436kB/s 
Installing collected packages: phonenumbers
  Found existing installation: phonenumbers 8.0.1
    Uninstalling phonenumbers-8.0.1:
      Successfully uninstalled phonenumbers-8.0.1
Successfully installed phonenumbers-8.1.0
$ ipython
Python 2.7.10 (default, Oct 23 2015, 19:19:21) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import phonenumbers, phonenumbers.geocoder

In [2]: phonenumbers.geocoder.description_for_number(phonenumbers.parse('+14422001212','US'),'en')
Out[2]: u'United States'

In [3]: 
```

**Comment by kputland** [comment]
None of the reported NPA have entries in the 8.1.0 geocoding data. Please see the check below, as well as a sample of two know NPA that are in the data.

```
In [4]: reported_NPA = ('1442', '1283', '1458', '1470', '1475', '1531', '1534', '1539', '1681', '1762')

In [5]: [ k for k in phonenumbers.geocoder.GEOCODE_DATA.keys() if k[:4] in reported_NPA ]
Out[5]: []

In [6]: [ k for k in phonenumbers.geocoder.GEOCODE_DATA.keys() if k[:4] in ('1303','1720') ]
Out[6]: 
['1303948',
 '1303745',
 '1303744',
 '1720570',
 '1720494',
...
 '1303470',
 '1303869',
 '1303739']

```

**Comment by kkeshava** [comment]
@kputland,
As mentioned earlier we will be updating the geocoding data for all the area codes mentioned by you. 
In comment#3 you have provided more specific geocoding details for area code 442. Would you please provide the evidence for it? So that we can implement the same way for other area codes as well.

**Comment by kputland** [comment]
Links to specifics were provided in each ticket.

**Comment by kputland** [comment]
1442 evidence from https://www.nationalnanpa.com/enas/coCodeReportUnsecured.do?reportType=7
[utilized.xlsx](https://github.com/googlei18n/libphonenumber/files/742977/utilized.xlsx)

**Comment by kkeshava** [comment]
Hi,
We auto-generate the US geocoding file and hence city-level details will be overwritten (but not state level details) if we manually add them. The state-level geocoding details are being added based on official NANPA site (https://www.nationalnanpa.com/) which are in line with the details provided by you. 
We have created a feature request in #1546 for adding more sources for auto-generating city-level US geocoding. Please follow that thread for updates.

**Comment by kkeshava** [comment]
Fixed in release 8.3.0.
